src/events.js

```diff
@@ -7,3 +10,49 @@ export const EVENT_READTHEDOCS_FLYOUT_SHOW = "readthedocs-flyout-show";
 export const EVENT_READTHEDOCS_FLYOUT_HIDE = "readthedocs-flyout-hide";
 export const EVENT_READTHEDOCS_ADDONS_DATA_READY =
   "readthedocs-addons-data-ready";
+
+/**
+ * Object to pass to user subscribing to `EVENT_READTHEDOCS_ADDONS_DATA_READY`.
+ *
+ * This object allows us to have a better communication with the user.
+ * Instead of passing the raw data, we pass this object and enforce them
+ * to use it in an expected way:
+ *
+ *   document.addEventListener(
+ *     "readthedocs-addons-data-ready",
+ *     function (event) {
+ *       // Initialize the class first
+ *       event.detail.initialize();
+ *
+ *       // Use the data when it's ready
+ *       event.detail.data().then((data) => {
+ *         console.log(data.projects.current.slug);
+ *       });
+ *     }
+ *   );
```

Comment:
I feel like when we trigger this event -- `data-ready` -- the data should be ready, and the underlying promise from `data()` should be resolved at that point.

It also feels odd to require two method calls, it seems like we can initialize and return the resolved data in one function or method.

That is:

```suggestion
 *   document.addEventListener(
 *     "readthedocs-addons-data-ready",
 *     function (event) {
 *       const data = event.detail.data();
 *     }
 *   );
```

This feels more expected.